### DISCUSSION TITLE
Open3D blank screen

### DISCUSSION THREAD

**Discussion by adam046** [isl-org/Open3D#4893] [Q&A]
I asked the same question on SO, but just for extra reach I'll also ask here. I'm using a ZED 2i camera and Open3D in python. The resolution mode is HD720 (1280*720) and the returned point cloud has this format: https://www.stereolabs.com/docs/depth-sensing/using-depth/ ; here is a code snippet:
```
# After instancing the camera, using .grab etc
zed.retrieve_measure(point_cloud, sl.MEASURE.XYZRGBA)
pc_data = point_cloud.get_data() # shape is (720, 1280, 4)

pc_data = np.reshape(pc_data, (-1, 4)) # shape is (921600, 4)

colors = pc_data[:, 3] # Taking the colors separately
pc_data = np.delete(pc_data, 3, axis=1) # Removing color column on 3D data

pc_colors = np.zeros_like(pc_data) # Preparing the point cloud's colors

# About this loop, see: https://www.stereolabs.com/docs/depth-sensing/using-depth/
# The RGBA values are stored in a single float, need to unpack
for i in range(colors.shape[0]):
    bytearr = bytearray(struct.pack("f", colors[i]))
    r = bytearr[0]
    g = bytearr[1]
    b = bytearr[2]
    a = bytearr[3]
    # print(f"RGB values: ({r}, {g} ,{b}), alpha: {a}")
    pc_colors[i, 0] = r / 255.0
    pc_colors[i, 1] = g / 255.0
    pc_colors[i, 2] = b / 255.0
    # No alpha

pcd = o3d.geometry.PointCloud()
pcd.points = o3d.utility.Vector3dVector(pc_data) # set points
pcd.colors = o3d.utility.Vector3dVector(pc_colors) # set colors

o3d.visualization.draw_geometries([pcd])
```
However, when I run it, the visualizer stays blank. And it works for others point clouds, e.g. the example ones from open3d. I don't know what is causing this, I omitted many debugging prints and it should all be working. I even tried opening the point cloud and then painting the voxels with open3d functions, but nothing. I'm sure I'm missing something, that I hope is trivial. Thanks for the help!

**Comment by Brynn-pl** [comment]
> I asked the same question on SO, but just for extra reach I'll also ask here. I'm using a ZED 2i camera and Open3D in python. The resolution mode is HD720 (1280*720) and the returned point cloud has this format: https://www.stereolabs.com/docs/depth-sensing/using-depth/ ; here is a code snippet:
> 
> ```
> # After instancing the camera, using .grab etc
> zed.retrieve_measure(point_cloud, sl.MEASURE.XYZRGBA)
> pc_data = point_cloud.get_data() # shape is (720, 1280, 4)
> 
> pc_data = np.reshape(pc_data, (-1, 4)) # shape is (921600, 4)
> 
> colors = pc_data[:, 3] # Taking the colors separately
> pc_data = np.delete(pc_data, 3, axis=1) # Removing color column on 3D data
> 
> pc_colors = np.zeros_like(pc_data) # Preparing the point cloud's colors
> 
> # About this loop, see: https://www.stereolabs.com/docs/depth-sensing/using-depth/
> # The RGBA values are stored in a single float, need to unpack
> for i in range(colors.shape[0]):
>     bytearr = bytearray(struct.pack("f", colors[i]))
>     r = bytearr[0]
>     g = bytearr[1]
>     b = bytearr[2]
>     a = bytearr[3]
>     # print(f"RGB values: ({r}, {g} ,{b}), alpha: {a}")
>     pc_colors[i, 0] = r / 255.0
>     pc_colors[i, 1] = g / 255.0
>     pc_colors[i, 2] = b / 255.0
>     # No alpha
> 
> pcd = o3d.geometry.PointCloud()
> pcd.points = o3d.utility.Vector3dVector(pc_data) # set points
> pcd.colors = o3d.utility.Vector3dVector(pc_colors) # set colors
> 
> o3d.visualization.draw_geometries([pcd])
> ```
> 
> However, when I run it, the visualizer stays blank. And it works for others point clouds, e.g. the example ones from open3d. I don't know what is causing this, I omitted many debugging prints and it should all be working. I even tried opening the point cloud and then painting the voxels with open3d functions, but nothing. I'm sure I'm missing something, that I hope is trivial. Thanks for the help!

same questions.I wonder if you have solved it and could you share it? Thank you very much
